src/tarray.jl

```diff
@@ -259,7 +283,7 @@ Base.:*(x::AbstractArray, y::TArray) = x * _get(y) |> localize
 Base.:*(x::TArray, y::AbstractArray) = _get(x) * y |> localize
 
 # broadcast
-Base.BroadcastStyle(::Type{TArray{T, N}}) where {T, N} = Broadcast.ArrayStyle{TArray}()
+Base.BroadcastStyle(::Type{TArray{T, N, A}}) where {T, N, A} = Broadcast.ArrayStyle{TArray}()
```

Comment:
Are there some problems with dropping the type information here? It seems it is not used anyway in the implementation of `broadcasted` so it should be fine, I assume?

Reply:
I am also curious about this: if I use `::Type{TArray}` or `::Type{TArray{T, N}}`, the broadcasting benchmark will consume about 10x as much time as using `::Type{TArray{T, N, A}}`.

Reply:
Maybe it falls back to some more expensive implementation in Julia base if the type is not concrete?

Reply:
But initial question was more about `ArrayStyle{TArray}` than `::Type{TArray{T,N,A}}` though.

Reply:
Oh, `ArrayStyle{TArray}` is just a singleton used to dispatch function calls to the propriety `Broadcast.broadcasted` implementation, so I think it's OK to drop the parametric types.

Reply:
According to the [documentation](https://docs.julialang.org/en/v1/manual/interfaces/#Broadcast-Styles) for `AbstractArray`s the preferred definition is
```suggestion
Base.BroadcastStyle(::Type{<:TArray}) = Broadcast.ArrayStyle{TArray}()
```
which is basically what you had above (and therefore should give similar performance).

Reply:
> Base.BroadcastStyle(::Type{<:TArray}) = Broadcast.ArrayStyle{TArray}()

I'd second the one from Julia documentation since it is more concise.